user_guide_src/source/incoming/auto_routing_improved.rst

```diff
@@ -0,0 +1,488 @@
+``App\Controllers\HelloWorld`` and executes ``getHello()`` method with passing
+``'1'`` as the first parameter.
+
+.. note:: A controller method that will be executed by Auto Routing (Improved)
+    needs HTTP verb (``get``, ``post``, ``put``, etc.) prefix like ``getIndex()``,
+    ``postCreate()``.
+
+.. note:: When a controller's short name matches the first segment of a URI, it
+    will be loaded.
+
+**************************
+Let's try it: Hello World!
+**************************
+
+Let's create a simple controller so you can see it in action.
+
+Create a Controller
+===================
+
+Using your text editor, create a file called **HelloWorld.php** in your
+**app/Controllers** directory, and put the following code in it.
+
+.. literalinclude:: auto_routing_improved/020.php
+
+.. important:: The file must be called **HelloWorld.php**. When you use Auto
+    Routing (Improved), controller class names MUST be CamelCase.
+
+You will notice that the ``HelloWorld`` Controller is extending the ``BaseController``.
+You can also extend the ``CodeIgniter\Controller`` if you do not need the functionality
+of the BaseController.
+
+The BaseController provides a convenient place for loading components and performing
+functions that are needed by all your controllers. You can extend this class in
+any new controller.
+
+.. important:: A controller method that will be executed by Auto Routing (Improved)
+    needs HTTP verb (``get``, ``post``, ``put``, etc.) prefix like ``getIndex()``,
+    ``postCreate()``.
+
+Check the Routes
+================
+
+You can check your routes with the ``spark routes`` command.
+
+.. code-block:: console
+
+    php spark routes
+
+If you did it right, you should see:
+
+.. code-block:: none
+
+    +-----------+-------------+------+---------------------------------------+----------------+---------------+
+    | Method    | Route       | Name | Handler                               | Before Filters | After Filters |
+    +-----------+-------------+------+---------------------------------------+----------------+---------------+
+    | GET(auto) | hello-world |      | \App\Controllers\HelloWorld::getIndex |                |               |
+    +-----------+-------------+------+---------------------------------------+----------------+---------------+
+
+See :ref:`routing-spark-routes` for the output details.
+
+Visit Your Site
+===============
+
+Now visit your site using a URL similar to this::
+
+    http://example.com/hello-world
+
+The system automatically translates URI with dashes (``-``) to CamelCase in the
+controller and method URI segments.
+
+For example, the URI ``sub-dir/hello-controller/some-method`` will execute the
+``SubDir\HelloController::getSomeMethod()`` method.
+
+If you did it right, you should see::
+
+    Hello World!
+
+****************************
+Examples of Controller Names
+****************************
+
+This is an valid controller name. Because ``App\Controllers\HelloWorld`` is CamelCase.
```

Comment:
Honestly, I didn't like using 'This is', it made the review difficult for me.

Reply:
I changed to `The following is`.